packages/web/components/Dashboard/Settings/Subscription/SubscriptionForm.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react'
+import { toast } from 'react-toastify'
+
+import Button, { ButtonVariant } from '@/components/Button'
+import { useConfirmationModal } from '@/components/Modals/ConfirmationModal'
+import { useTranslation } from '@/config/i18n'
+import {
+  MembershipSubscriptionPeriod,
+  UserWithSubscriptionFragmentFragment as UserType,
+  useUpdateSubscriptionRenewalMutation,
+} from '@/generated/graphql'
+import theme from '@/theme'
+import { formatLongDate } from '@/utils'
+import CardOnFile from './CardOnFile'
+import PaymentForm from './PaymentForm'
+import PaymentFormModal from './PaymentFormModal'
+
+type SubscriptionFormProps = {
+  user: UserType
+}
+
+const SubscriptionForm = ({ user }: SubscriptionFormProps) => {
+  const { t } = useTranslation('settings')
+  let subscriptionPlan: string | undefined
+  // (period) => string
+  if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Monthly) {
+    subscriptionPlan = '1 Month / £12'
+  } else if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Quarterly) {
+    subscriptionPlan = '3 Months / £30'
+  } else if (user?.membershipSubscription?.period === MembershipSubscriptionPeriod.Annualy) {
+    subscriptionPlan = '1 Year / £100'
+  }
+  const isCancelling = user.membershipSubscription?.cancelAtPeriodEnd
+  const [showPaymentForm, setShowPaymentForm] = useState(!user.isPremiumUser)
```

Comment:
for perf reasons, we should think about moving this field onto the subscription rather than the user. Shouldn't be too hard since it's a computed field, doesn't go back to the DB

Reply:
✅ Done!